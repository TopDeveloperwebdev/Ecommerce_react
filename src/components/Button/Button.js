import React from 'react';
import PropTypes from 'prop-types';

import style from './Button.sass';

const Button = ({ onClick, text }) => (
  <button className={style.button} onClick={onClick} >{text}</button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string,
};

Button.defaultProps = {
  text: '',
};

export default Button;
