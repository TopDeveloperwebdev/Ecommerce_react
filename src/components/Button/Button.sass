@import 'css/variables'


.button
  border: 1px solid $default-blue
  border-radius: $border-radius
  background-color: $default-blue
  color: #fff
  font-size: $font-size-sm
  padding: 7px 10px
  outline: none

  &:hover
    cursor: pointer
    background: $light-blue
    border-color: $light-blue
    box-shadow: inset 0 0 transparent, 0 0 0 3px $light-blue-shadow
    transition: .2s

