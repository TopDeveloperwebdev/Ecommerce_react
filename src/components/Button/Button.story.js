import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo, setDefaults } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import Button from './Button';

// addon-info
setDefaults({
  source: true,
});

storiesOf('Button', module)
  .add('With text', withInfo()(() => (
    <Button text="Button" />
  )))
  .add('onClick', withInfo()(() => (
    <Button onClick={action('clicked')} text="Click here" />
  )));
