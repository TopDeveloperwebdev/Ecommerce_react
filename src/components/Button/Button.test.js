import React from 'react';
import uid from 'uid';
import { shallow, mount } from 'enzyme';

import Button from './Button';

test('Should component render text', () => {
  const text = 'text';

  const wrapper = mount(<Button text={text} onClick={() => {}} />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.button').text()).toEqual(text);
});

test('Should component call onClick', () => {
  const text = 'text';
  const mockClick = jest.fn();

  const wrapper = mount(<Button text={text} onClick={mockClick} />);

  wrapper.simulate('click');
  expect(mockClick).toHaveBeenCalled();
});
