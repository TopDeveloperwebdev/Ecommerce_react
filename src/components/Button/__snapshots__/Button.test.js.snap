// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component render text 1`] = `
<Button
  onClick={[Function]}
  text="text"
>
  <button
    className="button"
    onClick={[Function]}
  >
    text
  </button>
</Button>
`;
